### ISSUE TITLE
Flutter commands delay on Windows and git-remote-https.exe process

### ISSUE THREAD

**Issue by ramioooz** [flutter/flutter#53733]
Good and CoronaVirus-Free day to everyone,

I have been experiencing a noticeable delay every time I run any flutter commands on windows 10. I tried to find the reason for that, open task manager and I run **flutter --version** in the windows command line. I noticed a new process starts every time I run flutter command, The process called: "**git-remote-https.exe**"

I tried to investigate it, I switched to another task manager for windows called "**Process Explorer**" and I open properties for "**git-remote-https.exe**" process to get more information about it. I took this screenshot :

![git-remote-https capture](https://user-images.githubusercontent.com/27827640/78111165-8d6e7280-73fc-11ea-9425-a926fa32e83c.PNG)

In the Command line section, you see that flutter is trying to get remote data from flutter.git repository, I don't know why this behavior is happening, considering that I have flutter SDK already downloaded at : D:\Flutter\sdk\ and I don't need to download any more data as far as I know.

I noticed that "**git-remote-https**" process downloads about **1.5MB** of data every time it runs and then terminates. I have to wait for the download to finish before getting the result of commands like "**flutter --version**" as if I am getting the result from a remote server.

I am using flutter version 1.15.17 on the beta channel. I have been experiencing this behavior only recently, for the last week

Thank you

**Comment by TahaTesser** [comment]
Hi @ramioooz 
can you please provide your `flutter doctor -v`?
Thank you

**Comment by ramioooz** [comment]
Hello @TahaTesser,

Thank you for your reply,

I run 'flutter doctor -v',
waited for sometime, about +2 minutes, _I guess for 'git-remote-https.exe' process to finish_
and here is the response : 

```
C:\Users\Rami>flutter doctor -v
[√] Flutter (Channel beta, v1.15.17, on Microsoft Windows [Version 10.0.18363.720], locale en-US)
    • Flutter version 1.15.17 at D:\Flutter\sdk
    • Framework revision 2294d75bfa (4 weeks ago), 2020-03-07 00:28:38 +0900
    • Engine revision 5aff311948
    • Dart version 2.8.0 (build 2.8.0-dev.12.0 9983424a3c)


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at D:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_SDK_ROOT = D:\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.43.2)
    • VS Code at C:\Users\Rami\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (3 available)
    • SM A520F   • 192.168.8.80:5555 • android-arm64  • Android 8.0.0 (API 26)
    • Chrome     • chrome            • web-javascript • Google Chrome 80.0.3987.162
    • Web Server • web-server        • web-javascript • Flutter Tools

• No issues found!
```

Hope this helps,
Thank you

**Comment by TahaTesser** [comment]
Hi @ramioooz 
Can you try this updating your git to latest version?
Does this happens with every command?  Did you notice same delay with `flutter run -v`? 
Thank you

**Comment by ramioooz** [comment]
Hello @TahaTesser ,

Yes, I am running the latest version of git for windows, as shown :
```
C:\Users\Rami>git --version
git version 2.26.0.windows.1

C:\Users\Rami>
```

and yes, I am getting the same delay for all flutter commands, including flutter run.
all flutter commands launch 'git-remote-https.exe' process in the background and I have to wait for it to download about 1.5MB of data before proceeding with the execution. 

Why do flutter commands connect to the internet? 

What is the data it needs to fetch from the internet before running the application?
 
If flutter commands most download some data, shouldn't it be caught in local disk instead of downloading over and over again with every run?

Thank you,

**Comment by jmagman** [comment]
Probably fetching tags, fixed by https://github.com/flutter/flutter/pull/53450.  Can you try on the dev channel?

**Comment by ramioooz** [comment]
> Probably fetching tags, fixed by #53450. Can you try on the dev channel?

Hello @jmagman,

I switched from package : 
https://storage.googleapis.com/flutter_infra/releases/beta/windows/flutter_windows_v1.15.17-beta.zip
to package : 
https://storage.googleapis.com/flutter_infra/releases/dev/windows/flutter_windows_1.18.0-dev.0.0-dev.zip 

and the problem solved. 

I think the issue was with beta package: flutter_windows_v1.15.17-beta.zip
now after switching to dev package, flutter commands don't fetch extra data from the internet and run with no delay.

Thank you

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/53450 is in 1.17.0 so it looks like that was it!  Thank you!
